convex/document.ts

```
@@ -222,16 +213,74 @@ export const getSearch=query({
+    icon: v.optional(v.string()),
+    isPublished: v.optional(v.boolean()),
+  },
+  handler:async (ctx,args)=>{
+    const identity=await ctx.auth.getUserIdentity();
+
+    if(!identity){
+      throw new Error("Unauthorized");
+    }
+
+    const userId=identity.subject;
+    const {id,...rest}=args;
```

ESLint

[error] 'id' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)